### ISSUE TITLE
Respect `prefers-color-scheme` of the underlying OS

### ISSUE THREAD

**Issue by jasikpark** [zed-industries/zed#4970]
### Check for existing issues

- [X] Completed

### Describe the feature

VSCode and many other apps support setting two different themes for light mode and dark mode, while respecting the underlying preference of the OS - this allows things like having dark mode at night and light mode during the day.

It'd be great to have support for this!

### If applicable, add mockups / screenshots to help present your vision of the feature

```json5
// Zed settings
//
// For information on how to configure Zed, see the Zed
// documentation: https://zed.dev/docs/configuring-zed
//
// To see all of Zed's default settings without changing your
// custom settings, run the `open default settings` command
// from the command palette or from `Zed` application menu.
{
    "theme": {
        "light": "One Light",
        "dark": "One Dark"
    },
    // "buffer_font_size": 15,
    // "language_overrides": {
    //     "JavaScript": {
    //         "format_on_save": {
    //             "external": {
    //                 "command": "npx prettier",
    //                 "arguments": [
    //                     "--stdin-filepath",
    //                     "{buffer_path}"
    //                 ]
    //             }
    //         }
    //     }
    // }
}
```

**Comment by MatthiasGrandl** [comment]
I added this feature in my own GPUI app [Loungy](https://github.com/MatthiasGrandl/Loungy/blob/main/src/theme.rs#L124) with the help of the [dark-light crate](https://docs.rs/dark-light). It works great although sadly it doesn't pick up system theme changes due to a [bug](https://github.com/frewsxcv/rust-dark-light/issues/29) on MacOS.

**Comment by texastoland** [comment]
> VSCode doesn't seem to be doing it automatically

It has worked for years with the following settings keys:

- `window.autoDetectColorScheme`
- `window.preferredDarkColorTheme`
- `window.preferredLightColorTheme`

I checked Alacritty and Rio but they both use winit's (complex) implementation ([Mac impl.](https://github.com/rust-windowing/winit/blob/97cfdd4b0972b3a743c1cfb0f085800a446f6324/src/platform_impl/macos/window_delegate.rs#L421-L445)).

**Comment by maxdeviant** [comment]
This is supported in Zed with the changes in https://github.com/zed-industries/zed/pull/7404.

**Comment by texastoland** [comment]
Wooow confirmed working 👏🏼

```json
"theme": {
  "light": "Ayu Light",
  "dark": "Ayu Dark"
},
```

**Comment by gregwebs** [comment]
I can confirm it works as well after updating Zed.

**Comment by maxdeviant** [comment]
Closing this issue since the feature is available in Zed 0.122.2.
